public/static/js/init.py

```diff
@@ -23,50 +24,77 @@ def __init__(self):
-        post_message({"cmd": "draw", "msg": json.dumps(json_map)})               
+
     def arcTo(self, x1, y1, x2, y2, radius):
-        json_map = {"action":"arc", "x1":x1, "y1":y1, "x2":x2, "y2":y2, "radius":radius}        
-        post_message({"cmd": "draw", "msg": json.dumps(json_map)}) 
+        json_map = {"action": "arc", "x1": x1, "y1": y1,
+                    "x2": x2, "y2": y2, "radius": radius}
+        post_message({"cmd": "draw", "msg": json.dumps(json_map)})
+
     def bezierCurveTo(self, cp1x, cp1y, cp2x, cp2y, x, y):
-        json_map = {"action":"bezierCurveTo", "cp1x":cp1x, "cp1y":cp1y, "cp2x":cp2x, "cp2y":cp2y, "x":x, "y":y}        
-        post_message({"cmd": "draw", "msg": json.dumps(json_map)}) 
+        json_map = {"action": "bezierCurveTo", "cp1x": cp1x,
+                    "cp1y": cp1y, "cp2x": cp2x, "cp2y": cp2y, "x": x, "y": y}
+        post_message({"cmd": "draw", "msg": json.dumps(json_map)})
+
     def moveTo(self, x, y):
-        json_map = {"action":"moveTo", "x":x, "y":y}
-        post_message({"cmd": "draw", "msg": json.dumps(json_map)})        
+        json_map = {"action": "moveTo", "x": x, "y": y}
+        post_message({"cmd": "draw", "msg": json.dumps(json_map)})
+
     def clearRect(self, x, y, width, height):
-        json_map = {"action":"clearRect", "x":x, "y":y, "width":width, "height":height}
-        post_message({"cmd": "draw", "msg": json.dumps(json_map)})        
+        json_map = {"action": "clearRect", "x": x,
+                    "y": y, "width": width, "height": height}
+        post_message({"cmd": "draw", "msg": json.dumps(json_map)})
+
     def lineTo(self, x, y):
-        json_map = {"action":"lineTo", "x":x, "y":y}
-        post_message({"cmd": "draw", "msg": json.dumps(json_map)})  
+        json_map = {"action": "lineTo", "x": x, "y": y}
+        post_message({"cmd": "draw", "msg": json.dumps(json_map)})
+
     def fillText(self, text, x, y, maxWidth="", clearCanvas=False):
-        json_map = {"action":"fillText", "text":text, "x":x, "y":y, "maxWidth":maxWidth, "clearCanvas": clearCanvas}
+        json_map = {"action": "fillText", "text": text, "x": x,
+                    "y": y, "maxWidth": maxWidth, "clearCanvas": clearCanvas}
         post_message({"cmd": "draw", "msg": json.dumps(json_map)})
+
     def strokeText(self, text, x, y, maxWidth="", clearCanvas=False):
-        json_map = {"action":"strokeText", "text":text, "x":x, "y":y, "maxWidth":maxWidth, "clearCanvas": clearCanvas}
-        post_message({"cmd": "draw", "msg": json.dumps(json_map)})      
+        json_map = {"action": "strokeText", "text": text, "x": x,
+                    "y": y, "maxWidth": maxWidth, "clearCanvas": clearCanvas}
+        post_message({"cmd": "draw", "msg": json.dumps(json_map)})
+
+    def check_key(self, key_code):
+        return js.workerCheckKeyDown(key_code)
```

Comment:
this is new!

Reply:
Works well (aside from minor flicker as you say). The flicker won't be a big issue apart from moving canvas games anyway and mostly I was thinking about simpler more static things like e.g. noughts and crosses although that does raise the question about a click event: could we add that into a different shared buffer and check for click?